### PR TITLE
Update Applications Table

### LINKED ISSUE #1851: Update "Applications" Table
### Epic

#1689

### Description

Screenshot 2023-03-16 at 14 51 05

With the introduction of multiple instance per project, and the addition of "Application" as an entity server-side, we will now have access to things like the Application's "name". With this, we should update the "Applications" view in order to display the relevant hierarchy of Applications > Instances.

### Which customers would this be availble to

None

### Acceptance Criteria

- [ ] Table lists all Applications with their relevant names
- [ ] Each Application as a nested set of rows with their "Instances" listed.

### PR DESCRIPTION
**Pull request by joepavitt**
## Description

Correct me if this isn't the branch to be PRing back into @Pezmc. This also includes a merge of `main` so I could get the other UI work I'd done, hence the ridiculous number of files changed and commits, not sure how best to coordinate branches here, I can continue to pull updates from your [feat-1735-expose-applications](https://github.com/flowforge/flowforge/tree/feat-1735-expose-applications) branch and then this PR _could_ be the primary merge to `main`?

Updates:
- [x] Remove Team > Overview Page
- [x] Rework Applications table to show nested Instances

Still to do:
- [x] Update E2E Tests
- [x] Possible change of layout of table (see comments below)
- [x] Cannot render the "Open Editor" button as that is currently missing from the instances list for the `/applications/<application-id>/instances` API endpoint.

## Related Issue(s)

Closes #1851 

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/flowforge/flowforge/blob/main/CONTRIBUTING.md)
 - [x] Suitable unit/system level tests have been added and they pass

### CODE REVIEW

**Comment by joepavitt** [comment]
First pass, in order to align with the design resented [here](https://github.com/flowforge/flowforge/issues/1851#issue-1627679009):

Screenshot 2023-03-30 at 16 45 13

This doesn't look very good when it's just a 1:1 Application > Instance, although I don't mind how it is in the design linked.

As such, going to iterate on this to: https://github.com/flowforge/flowforge/issues/1851#issuecomment-1490492962

**Comment by joepavitt** [comment]
Dev'd up the alternative view, it uses up a lot more vertical space, but think it breaks down the hierarchy better. This is quit an unlikely scenario, where they have 6 different applications, vertical space is reduced for multiple instance within a given application due to where the padding lives.

In both scenarios, the duplication of Instance/Application name makes it trickier to digest too

https://user-images.githubusercontent.com/99246719/228917164-94d52626-9d51-4deb-be94-b5c88bc5f8bb.mov

**Comment by Pezmc** [comment]
> In both scenarios, the duplication of Instance/Application name makes it trickier to digest too

This shouldn't apply any more, create new instances in your applications and delete the ones with the duplicate names (and/or create new applications).

**Comment by joepavitt** [comment]
> This shouldn't apply any more, create new instances in your applications and delete the ones with the duplicate names (and/or create new applications).

Appreciate that - didn;t see an API (exposed in the `application.js` service to update name of an existing Application? If there is an API for it, I'll just add the function to the service and functionality into UI

**Comment by Pezmc** [comment]
Renaming is not available yet, create new ones! The API is there for it though, see the application route on the backend.

**Comment by joepavitt** [comment]
If the API is there (I'll go digging in a second) - I'll just add it front-end - 10 mins of work.

**Comment by joepavitt** [comment]
All development work now done - just tests remaining which I will get to next week (Tuesday)

**Comment by Pezmc** [comment]
@joepavitt what is the other branches merged to me, I will do a rebase and then review this branch

**Comment by joepavitt** [comment]
This has a merge from `main` and regular `merge` from your branch that this merges to. A lot of the difference here will come from the diff between `main` and your branch, so may be worth pulling latest `main` into yours to get the correct difference here?

Alternatively, we can switch this PR to point to `main` and merge your PR independently (and first)

File: frontend/src/components/DevicesBrowser.vue [hunk 1]
```diff
@@ -226,20 +224,21 @@ export default {
...
-                        component: {
-                            is: markRaw(ApplicationLink),
-                            map: {
-                                id: 'project.id',
-                                name: 'project.name'
-                            }
-                        }
-                    })
+                    ...statusColumns
+                    // {
```
Comment (Pezmc) [inline frontend/src/components/DevicesBrowser.vue:228]:
```suggestion
                    // TODO Restore application
                    // {
```

File: frontend/src/pages/team/Applications.vue [hunk 2]
```diff
@@ -1,55 +1,100 @@
...
+        <ff-loading v-if="loading" message="Loading Applications..." />
+        <template v-else-if="!loading && Array.from(applications.values()).length > 0">
+            <ul class="ff-applications-list">
+                <li v-for="application in Array.from(applications.values())" :key="application.id">
+                    <div class="ff-application-list--app" @click="openApplication(application)" data-action="view-application">
+                        <span class="flex justify-center"><TemplateIcon class="ff-icon text-gray-600" />{{ application.name }}</span>
+                        <label class="italic text-gray-400 text-sm">
+                            {{ application.instances.length }} Instance{{ application.instances.length === 1 ? '' : 's' }}
+                        </label>
+                        <!-- <ff-kebab-menu data-el="application-options">
```
Comment (Pezmc) [inline frontend/src/pages/team/Applications.vue:27]: ✂️

**Review by Pezmc — APPROVED** [review]
Addressed my comments since Joe is out

**Pull request merged**